File: 2346-largest-3-same-digit-number-in-string/2346-largest-3-same-digit-number-in-string.py
class Solution(object):
    def largestGoodInteger(self, num):
        """
        :type num: str
        :rtype: str
        """
        maxDigit = None
        n = len(num)

        for i in range(n-2):
            if num[i] == num[i+1] and num[i] == num[i+2]:
                if maxDigit == None or maxDigit < ord(num[i])-ord('0'):
                    maxDigit = ord(num[i])-ord('0')

        ans = ""
        if maxDigit is not None:
            for i in range(3):
                ans += str(maxDigit)

        return ans

File: 2346-largest-3-same-digit-number-in-string/2346-largest-3-same-digit-number-in-string.cpp
class Solution {
public:
    string largestGoodInteger(string num) {
        // Assign 'maxDigit' to the NUL character (smallest ASCII value character)
        char maxDigit = '\0';

        // Iterate on characters of the num string.
        for (int index = 0; index <= num.size() - 3; ++index) {
            // If 3 consecutive characters are the same,
            // store the character in 'maxDigit' if bigger than what it already stores.
            if (num[index] == num[index + 1] && num[index] == num[index + 2]) {
                if (num[index] > maxDigit) {
                    maxDigit = num[index];
                }
            }
        }

        // Use regular if-else instead of ternary operator
        if (maxDigit == '\0') {
            return "";
        } else {
            return string(3, maxDigit);
        }
    }
};